python/ray/tests/test_draining.py

```diff
@@ -434,6 +434,56 @@ def ping(self):
             assert "the actor's node was preempted: " + drain_reason_message in str(e)
 
 
+def test_drain_node_actor_restart(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=1, resources={"head": 1})
+    ray.init(address=cluster.address)
+
+    gcs_client = GcsClient(address=ray.get_runtime_context().gcs_address)
+
+    max_restarts = 1
+
+    @ray.remote(max_restarts=max_restarts)
+    class Actor:
+        def get_node_id(self):
+            return ray.get_runtime_context().get_node_id()
+
+    # Prepare the first worker node for the actor.
+    cur_worker = cluster.add_node(num_cpus=1, resources={"worker": 1})
+    cluster.wait_for_nodes()
+
+    actor = Actor.options(num_cpus=0, resources={"worker": 1}).remote()
+
+    node_id = ray.get(actor.get_node_id.remote())
+    assert node_id == cur_worker.node_id
+
+    # Loop max_restarts + 1 times because this test is to ensure that the actor can be restarted
+    # regardless max_restarts if its worker node is drained.
+    for _ in range(max_restarts + 1):
```

Comment:
no need for the looping logic since `max_restarts` is set to 1

I would also add a sanity check that if we then remove the node when it _isn't_ draining, it does not get restarted